### ISSUE TITLE
calculate measures fails

### ISSUE THREAD

**Issue by r3tlove** [blazek/lrs#10]
We have successfully used this wonderful plugin in the recent past. However, something has changed (see notes below) with QGIS/OSGEO4W and now we are no longer able to make the process work. We have a sample dataset (see link below) that worked in the past and now errors out on the calculate measures function. the process halts and errors at 24%.

**sample data set here**
http://projects.region3dfg.org/temporary/lrs/

**on installing the plugin i get..**

2016-08-23T17:02:05 1 warning:C:/Users/jack/.qgis2/python/plugins\lrs\lrsplugin.py:44: DeprecationWarning: QgsMapCanvas.mapRenderer() is deprecated
self.mapRenderer = iface.mapCanvas().mapRenderer()
File "C:/Users/jack/.qgis2/python/plugins\lrs\__init__.py", line 53, in classFactory
return LrsPlugin(iface)

**on loading the plugin i also get**

traceback: File "<string>", line 1, in <module>
File "C:/OSGeo4W/apps/qgis/./python\qgis\utils.py", line 333, in startPlugin
plugins[packageName] = package.classFactory(iface)

**On running the Calibrate and the calculate measures functions i get the following and the process locks up as described above.**

Python Errors:
2016-08-23T14:05:28 1 Traceback (most recent call last):
File "C:/Users/*/.qgis2/python/plugins\lrs\lrsdockwidget.py", line 973, in calculateMeasures
outputFeature[field.name()] = feature[field.name()]
KeyError: 'ageofdgpsd'

Python Warning:
2016-08-23T17:05:39 1 warning:C:/Users/jack/.qgis2/python/plugins\lrs\lrsplugin.py:44: DeprecationWarning: QgsMapCanvas.mapRenderer() is deprecated
self.mapRenderer = iface.mapCanvas().mapRenderer()
traceback: File "<string>", line 1, in <module>
File "C:/OSGeo4W/apps/qgis/./python\qgis\utils.py", line 333, in startPlugin
plugins[packageName] = package.classFactory(iface)
File "C:/Users/jack/.qgis2/python/plugins\lrs\__init__.py", line 53, in classFactory
return LrsPlugin(iface)
2016-08-23T17:07:49 1 warning:C:/Users/jack/.qgis2/python/plugins\lrs\lrs.py:301: DeprecationWarning: QgsDistanceArea.measure() is deprecated
length = self.distanceArea.measure( feature.geometry() )
traceback: File "C:/Users/jack/.qgis2/python/plugins\lrs\lrsdockwidget.py", line 487, in generateLrs
self.lrs.calibrate()
File "C:/Users/jack/.qgis2/python/plugins\lrs\lrs.py", line 238, in calibrate
self.registerLines()

**operational environment**
LRS Plugin Installed version: 0.3.6 (in C:\Users\*\.qgis2\python\plugins\lrs)

tested on...
useing OSGEO4W package installer
QGIS version 2.16.1-Nødebo
QGIS code revision 8545b3b
Compiled against Qt 4.8.6
Running against Qt 4.8.6
Compiled against GDAL/OGR 2.1.0
Running against GDAL/OGR 2.1.1
Compiled against GEOS 3.5.0-CAPI-1.9.0
Running against GEOS 3.5.0-CAPI-1.9.0 r4084
PostgreSQL Client Version 8.3.10
SpatiaLite Version 4.3.0
QWT Version 6.0.2
PROJ.4 Version 492
QScintilla2 Version 2.8.4

Windows 7 Enterprise 64 bit Service Pack 1
and
Windows 10 32bit

Previous success:
QGIS 2.10 with LRS 0.3.6 worked Prior to August 12, 2016.
Starting August 12, 2016 QGIS would crash after successfully completing LRS Events.
When saving the created Events layer QGIS would not save all records. Problems with exporting fields with lengths in excess of 254 characters. This field (LRS Errors)was all null.
QGIS would also crash when trying to render a new View after successfully running the LRS Events routine.
8/15/2016 upgraded to QGIS 2.16 and LRS worked again. Previous errors went away.
8/17/2016 ran setup utility again and now LRS errors out in Measure routine described above. Confirmed on two different computers with same setup.

**Comment by blazek** [comment]
Let us concentrate on QGIS 2.16. I tried to reproduce the error with 2.16.1 and your dataset, unfortunately I haven't got the error in measures calculation. 

The error you reported:
> File "C:/Users/*/.qgis2/python/plugins\lrs\lrsdockwidget.py", line 973, in calculateMeasures
> outputFeature[field.name()] = feature[field.name()]
> KeyError: 'ageofdgpsd'

looks strange, because it is writing all fields of all measured points 2016-08-16_MW_ALL and suddenly it fails on 'ageofdgpsd'. It is also strange that it worked 8/15/2016 and stopped 8/17/2016.

What do you mean by "8/17/2016 ran setup utility again"? Which setup utility?

Can you try to delete the column 'ageofdgpsd' from 2016-08-16_MW_ALL?

**Comment by r3tlove** [comment]
the error related to the field 'ageofdgpsd' is possibly due to the field being of type integer64, (which before this issue i had never seen a field of that type). i'm not sure exactly which program (upstream) is writing out that field type but we will try to avoid using it in the future as "integer64' would seem to be unnecessary for our needs. deleting the field seems to eliminate that point of failure but we will compare it to our previous work to make sure. 

'ran setup utility again'...all that was meant there was that we reran osgeo4w and updated the version of qgis that we were using. 

so i guess the depreciation warnings are not related to the program locking up, its looking like its just the integer64 issue.

**Comment by blazek** [comment]
Can I close the issue?

If it is a integer64 problem then it is probably a general bug in qgis worth to report in qgis issues.

**Comment by r3tlove** [comment]
well i am not knowledgeable enough to say if it is a qgis issue or a issue with LRS. is there a way in the code to check if the field is integer64 and then deal with it diffrently? or is this an area where qgis was suppose to handle it seamlessly (abstracted?).

**Comment by blazek** [comment]
Integer64 was introduced in GDAL 2, after GDAL upgrade I was able to reproduce the error. The problem was that temporary layer was created using uri generated by memory provider, which the provider itself was not able to parse (which could be also improved in QGIS).

I have changed how the memory layers are created, so that it work also with Integer64. The fix is in release 0.3.7 which I uploaded to QGIS plugins repository.
